src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToGenerateSupportedTargetFrameworkAlias.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using FluentAssertions;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Commands;
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using System.Xml.Linq;
+using System.Linq;
+using System.Collections.Generic;
+using NuGet.Frameworks;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToGenerateSupportedTargetFrameworkAlias : SdkTest
+    {
+        public GivenThatWeWantToGenerateSupportedTargetFrameworkAlias(ITestOutputHelper log) : base(log)
+        {}
+
+        [Theory]
+        [InlineData("netcoreapp3.1")]
+        [InlineData("net5.0")]
+        [InlineData("net6.0")]
+        [InlineData("netstandard2.1")]
+        [InlineData("net48")]
+        public void It_generates_supported_target_framework_alias_items(string currentTargetFramework)
+        {
+            TestTargetFrameworkAlias(currentTargetFramework, propertySetToTrue: null, new[]
+                {
+                    "netcoreapp3.0",
+                    "netcoreapp3.1",
+                    "net5.0",
+                    "net6.0",
+                    "netstandard2.0",
+                    "netstandard2.1",
+                    "net471",
+                    "net48"
+                });
+        }
+
+        [WindowsOnlyTheory]
+        [InlineData("net5.0-windows")]
+        [InlineData("net6.0-windows")]
+        public void It_generates_supported_target_framework_alias_items_when_targeting_windows(string currentTargetFramework)
+        {
+            TestTargetFrameworkAlias(currentTargetFramework, propertySetToTrue: null, new[]
+                {
+                    "netcoreapp3.0",
+                    "netcoreapp3.1",
+                    "net5.0-windows7.0",
+                    "net6.0-windows7.0",
+                    "netstandard2.0",
+                    "netstandard2.1",
+                    "net471",
+                    "net48"
+                });
+        }
+
+        [WindowsOnlyTheory]
+        [InlineData("net5.0", "UseWpf")]
+        [InlineData("net5.0", "UseWindowsForms")]
```

Comment:
This should be `net5.0-windows`, and I think there's a bug that will cause the resulting target framework to be `net5.0-windows-windows` or something in that case.